test/iron-fit-behavior.html

```diff
@@ -403,6 +404,303 @@
+          assert.isFalse(intersects(rect, fitRect), 'not visible');
+        });
+
+        test('top with offset', function() {
+          el.verticalAlign = 'top';
+          el.verticalOffset = 10;
+          el.refit();
+          var rect = el.getBoundingClientRect();
+          assert.equal(rect.top, parentRect.top + 10, 'top ok');
+          assert.equal(rect.height, elRect.height, 'no cropping');
+        });
+
+        test('top, noOverlap', function() {
+          el.verticalAlign = 'top';
+          el.noOverlap = true;
+          el.refit();
+          var rect = el.getBoundingClientRect();
+          assert.isFalse(intersects(rect, parentRect), 'no overlap');
+          assert.equal(rect.height, elRect.height, 'no cropping');
+        });
+
+        test('bottom', function() {
+          el.verticalAlign = 'bottom';
+          el.refit();
+          var rect = el.getBoundingClientRect();
+          assert.equal(rect.bottom, parentRect.bottom, 'bottom ok');
+          assert.equal(rect.height, elRect.height, 'no cropping');
+        });
+
+        test('bottom, update max-height', function() {
+          parent.style.top = (100 - parentRect.height) + 'px';
+          el.verticalAlign = 'bottom';
+          el.refit();
+          var rect = el.getBoundingClientRect();
+          assert.equal(rect.bottom, 100, 'bottom ok');
+          assert.equal(rect.height, 100, 'height ok');
+        });
+
+        test('bottom, handle min-height', function() {
+          parent.style.top = (100 - parentRect.height) + 'px';
+          el.verticalAlign = 'bottom';
+          el.style.minHeight = '200px';
+          el.refit();
+          var rect = el.getBoundingClientRect();
+          assert.isFalse(intersects(rect, fitRect), 'not visible');
+        });
+
+        test('bottom with offset', function() {
+          el.verticalAlign = 'bottom';
+          el.verticalOffset = 10;
+          el.refit();
+          var rect = el.getBoundingClientRect();
+          assert.equal(rect.bottom, parentRect.bottom - 10, 'bottom ok');
+          assert.equal(rect.height, elRect.height, 'no cropping');
+        });
+
+        test('bottom, noOverlap', function() {
+          el.verticalAlign = 'bottom';
+          el.noOverlap = true;
+          el.refit();
+          var rect = el.getBoundingClientRect();
+          assert.isFalse(intersects(rect, parentRect), 'no overlap');
+          assert.equal(rect.height, elRect.height, 'no cropping');
+        });
+
+        test('auto --> top', function() {
+          el.verticalAlign = 'auto';
+          el.refit();
+          var rect = el.getBoundingClientRect();
+          assert.equal(rect.top, parentRect.top, 'auto aligned to top');
+          assert.equal(rect.height, elRect.height, 'no cropping');
+        });
+
+        test('auto --> bottom', function() {
+          // This would cause a cropping of the element, so it should automatically
+          // align to the bottom to avoid it.
```

Comment:
This is just a code style suggestion, no real action required: when writing tests, consider putting content like this comment in the name of the test. This allows someone running the test to immediately know what passed or failed. For example:

```javascript
test('cropping the element when vertical align is auto aligns it to the bottom', ...);
```